### PR TITLE
Task/Consider collateral price in calculator

### PR DESCRIPTION
**Pull request by antoncoding**
# Task: Update Calculator

## High Level Description

Add logic to deal with options that have different strike/ underlying than collateral.
Add one more function called `_getToCollateralRate` that deal with this conversion rate

### For put:
* conversion only happen when strike != collateral
* eg, underlying = ETH, strike = USDC, collateral = cUSDC
* in this case, the output of `_getMarginRequired` will be denominated in strike (USDC)
* the output of `_getToCollateralRate` will tell us 1 USDC = ? cUSDC. Let's say ? = 50
* assume output of `_getMarginRequired` is 50 USDC, the real required collateral is 2500 USD.

### For call 
* conversion only happen when underlying != collateral
* eg, underlying = ETH, strike = USDC, collateral = cETH
* in this case, the output of `_getMarginRequired` will be denominated in strike (ETH)
* the output of `_getToCollateralRate` will tell us 1 ETH = ? cETH. Let's say ? = 50
* assume output of `_getMarginRequired` is 1 ETH, the real required collateral is 50 cETH 
### Code

- [x] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [x] Is your code up to date with the spec? 
- [x] Have you added your tests to the testing doc?

### CODE REVIEW

File: contracts/MarginCalculator.sol [hunk 1]
```diff
@@ -72,13 +73,22 @@ contract MarginCalculator is Initializable {
         // Vault contains no short tokens: return collateral value.
         if (_isEmptyAssetArray(_vault.shortOtokens)) return (SignedConverter.intToUint(collateralAmount.value), true);
 
+        // get required margin, denominated in strike or underlying asset
```
Comment (aparnakr) [inline contracts/MarginCalculator.sol:76]: this is a little confusing that for puts it is denominated in strike and call in underlying but the same function returns both.

File: contracts/MarginCalculator.sol [hunk 2]
```diff
@@ -72,13 +73,22 @@ contract MarginCalculator is Initializable {
...
         FixedPointInt256.FixedPointInt memory marginRequirement = _getMarginRequired(_vault);
```
Comment (aparnakr) [inline contracts/MarginCalculator.sol:77]: can we split this up for puts and calls? 

have it as marginRequiredInStrikeAsset = .... 
and marginRequiredInUnderlyingAsset = ... 

if not the naming is a little confusing

Comment (aparnakr) [reply contracts/MarginCalculator.sol:77]: Alternatively can all this happen in the getMarginRequired function?

**Review by aparnakr — APPROVED** [review]
Looks good, I think the exchange rate calcluation / nomenclature can be simplified. Left some ideas, happy to brainstorm more. It took me a bit to understand, so it might take someone who doesn't know the system inside out a little longer to understand.

the terminology margin and collateral and swapping the two was confusing. Its okay to merge for now and simplify it later (if this is urgent).

**Review by haythem96 — APPROVED** [review]

**Comment by haythem96** [comment]
Let's merge this after finishing Controller.

**Pull request merged**